### PR TITLE
Remove WordPress data types since they are now bundled

### PR DESCRIPTION
**Pull request by noahtallen**
### Changes
This removes the `@wordpress/data` types from DT since they are now bundled (as of version 7 of the package.) (See https://github.com/WordPress/gutenberg/pull/43315)

We might need to make some more updates to the other WordPress DT types to make them compatible with this change.

### Template:

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by noahtallen** [comment]
Also cc @adamziel

**Comment by noahtallen** [comment]
The errors were because some of the other WordPress packages had an implicit dependency on WordPress data. (Now this dependency is explicit)

File: types/wordpress__data-controls/index.d.ts [hunk 1]
```diff
@@ -53,13 +52,12 @@ export function apiFetch(options: APIFetchOptions): unknown;
  * @returns An object for registering the default controls with the store.
  */
 export const controls: {
-    API_FETCH: (action: Action) => Promise<any>;
-    DISPATCH: (action: Action) => void;
-    SELECT: (action: Action) => any;
```
Comment (noahtallen) [inline types/wordpress__data-controls/index.d.ts:58]: These are no longer provided per the source code: https://github.com/WordPress/gutenberg/blob/721cb342060a5f6ddf34f25e5379e6d8a4c806ae/packages/data-controls/src/index.js

File: types/wordpress__data-controls/index.d.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
-// Type definitions for @wordpress/data-controls 2.2
+// Type definitions for @wordpress/data-controls 2.30
 // Project: https://github.com/WordPress/gutenberg/tree/master/packages/data-controls/README.md
 // Definitions by: Derek Sifford <https://github.com/dsifford>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.7
 
 import { APIFetchOptions } from '@wordpress/api-fetch';
-import { Action } from '@wordpress/data';
```
Comment (noahtallen) [inline types/wordpress__data-controls/index.d.ts:8]: This didn't really give us anything useful, so we might as well remove the dependency on wp data

File: types/wordpress__data-controls/index.d.ts [hunk 3]
```diff
@@ -1,11 +1,10 @@
-// Type definitions for @wordpress/data-controls 2.2
+// Type definitions for @wordpress/data-controls 2.30
```
Comment (noahtallen) [inline types/wordpress__data-controls/index.d.ts:1]: This matches the latest published version of the package.

File: types/wordpress__data-controls/index.d.ts [hunk 4]
```diff
@@ -53,13 +52,12 @@ export function apiFetch(options: APIFetchOptions): unknown;
  * @returns An object for registering the default controls with the store.
  */
 export const controls: {
-    API_FETCH: (action: Action) => Promise<any>;
-    DISPATCH: (action: Action) => void;
-    SELECT: (action: Action) => any;
+	API_FETCH: ({request}: {request: APIFetchOptions}) => Promise<unknown>,
```
Comment (DangerBotOSS) [inline types/wordpress__data-controls/index.d.ts:55]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## wordpress__data-controls ([<kbd>unpkg</kbd>](https://unpkg.com/browse/wordpress__data-controls@latest/))
was missing the following properties:
1. syncSelect
## wordpress__edit-post ([<kbd>unpkg</kbd>](https://unpkg.com/browse/wordpress__edit-post@latest/))
was missing the following properties:
1. store
1. store
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8b5f9223dd3acd78faeb2e2d9ffc562219995ac4
</p>

**Comment by noahtallen** [comment]
- syncSelect is deprecated, so not really worth typing at this point
- edit post should get the store typed in another PR :)

File: types/wordpress__viewport/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (noahtallen) [inline types/wordpress__viewport/package.json:1]: Git is just wrong here -- we're deleting the first directory, and adding a completely different package.json in another directory :)

**Review by sheetalkamat — APPROVED** [review]

**Comment by noahtallen** [comment]
Ready to merge

**Comment by marekdedic** [comment]
Hey, you beat me to it! (#63638) :D Thanks!

**Comment by noahtallen** [comment]
> Hey, you beat me to it! (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63638) :D Thanks!

Very nice! The next step is getting the stores actually compatible with newer WP data versions. I'm trying to publish more build-types from Gutenberg itself to avoid the extra maintenance burden over here! (e.g. https://github.com/WordPress/gutenberg/pull/49647)

**Comment by marekdedic** [comment]
That's great, I have been slowly and happily updating and removing all the `@types/wordpress__...` packages when possible :D Hope to one day remove them all... :)

**Pull request merged**